src/features/api-keys/lib.ts

```diff
@@ -221,3 +221,42 @@ export const createGenericApiKey = async (
 		},
 	);
 };
+
+export const isApiKeyWithRole = async (
+	key: string,
+	roleName: string,
+	tx?: Tx,
+) => {
+	const role = await api.Auth.get({
+		resource: 'role',
+		passthrough: { tx, req: permissions.root },
+		id: {
+			name: roleName,
+		},
+		options: {
+			$select: '*',
```

Comment:
It'd be better to just select a single property, eg
```suggestion
			$select: 'id',
```

Reply:
Looks like I forgot a debug * in there :(